src/Config/RuntimeConfigLoader.cs

```diff
@@ -30,25 +30,77 @@ public abstract class RuntimeConfigLoader
     // state in place of using out params.
     public RuntimeConfig? RuntimeConfig;
 
+    public RuntimeConfigLoader(HotReloadEventHandler<HotReloadEventArgs>? handler = null, string? connectionString = null)
+    {
+        _handler = handler;
+        _connectionString = connectionString;
+    }
+
+    protected virtual void QueryManagerFactoryOnConfigChangedEvent(HotReloadEventArgs args)
+    {
+        _handler?.OnConfigChangedEvent(nameof(QueryManagerFactoryOnConfigChangedEvent), this, args);
+    }
+
```

Comment:
RE: the `QueryManagerFactoryOnConfigChangedEvent` prefix `QueryManagerFactory`.

`nameof(QueryManagerFactoryOnConfigChangedEvent)` is resolving the symbol name for `RuntimeConfigLoader.QueryManagerFactoryOnConfigChangedEvent` so you'd need to keep naming in sync in two locations (I'm assuming this is due to the ciruclar dependency issue between the Config and Core projects?):
1. `RuntimeConfigLoader.QueryManagerFactoryOnConfigChangedEvent`
2. `QueryManagerFactory.QueryManagerFactoryOnConfigChangedEvent`

Reply:
Right, if we add a reference to the config proj to pickup the names from the classes in core we get that same cycle issue. I couldnt find a more graceful way to get the names to match aside from just maintaining two names that are the same in the different classes in sync.

Reply:
if you had a class in the config project such as "EventNames", "Events", or "DabConfigEvents" with constants for the event name, that would be usable from the Core project because the config project is already imported. 

Then in the Core project code that subscribes to events, you could use "eventName = DabConfigEvents.MY_EVENT_NAME_1" and then drop the redundant prefix in MyFactory::MyFactoryEventOnChange.
Both the config and core projects would reference one event name constant so any updates in the constant would propagate to both projects the event name is used and nothing goes out of sync.

Reply:
Ya that makes sense, will add that in, thanks.